### ISSUE TITLE
Support ORCA 5.0

### ISSUE THREAD

**Issue by danielhollas** [pzarabadip/aiida-orca#26]
Hi @pzarabadip,

I am currently attending the aiida tutorial, and was thinking that as a final project I could work on updating this plugin to support ORCA 5.0, which was just released last week. Let me know what you think. Thanks!

**Comment by pzarabadip** [comment]
Hi @danielhollas 
Thanks for your interest in `aiida-orca` plugin.
That's a great suggestion. I was thinking about it and went through the `ORCA 5.0` manual to get a sense of changes. To the best of my knowledge about `ORCA 5.0`, there are couple of backward incompatibilities and mostly change of behavior on the default keywords. I'm currently not setting any of those defaults in this plugin to be affected but it would be nice to have mechanism in place to deal with them properly. I have following suggestion (if I may) for your kind contribution:

1. `Parser`: I'm currently using `cclib` as the parser for the plugin. I'm not sure how robust it is in parsing `ORCA 5.0` output files. It may worth doing some tests to see if it breaks or the output of parser is not formatted as we want. 
2. I think we need to have support for both `ORCA 4.x` and `ORCA 5.x` (at least for another year). In case of needing different parsing mechanism, we need to have a mechanism in place to identify the `ORCA` version and consequently sets proper parser or input validator, and etc. 
3. One important thing that needs to change in the plugin is the usage of protected classes, e.g.:
https://github.com/pzarabadip/aiida-orca/blob/693c21706f22f97f4be74eb68ef6e9f76cd79acc/aiida_orca/parsers/__init__.py#L77-L80
It would be nice to fix them.

Please let me know, if I could be any help. 

Good luck in `aiida-tutorial`, it's such a nice and useful event. 

Cheers,
Pezhman

**Comment by danielhollas** [comment]
@pzarabadip awesome, thank you so much for the detailed info. Let's see how far I can get. The hackathon is on Friday. 

To your point 2, definitely agree. I'll ask the aiida folks if there is some canonical way of handling code versions.

It might also be nice to bring the plugin with Aiida 2.0, in case it is currently not compatible (but that's just me being ambitious now, let's see... :-) )

**Comment by pzarabadip** [comment]
@danielhollas You're highly welcome. 
We may have a call on Friday to discuss and plan changes. 

I totally agree with having `AiiDA 2.0` compatible version of plugin now before it gets too late.
Let's see how it would go till Friday. 
Cheers,

**Comment by danielhollas** [comment]
@pzarabadip just a short report that `cclib` unfortunately seems to choke on the ORCA-5.0 output. Well, at least on the output of TDDFT calculation, I did not yet test anything else. Here's the stack trace:

```python
| [26|OrcaCalculation|on_except]: Traceback (most recent call last):
 |   File "/home/hollas/anaconda3/envs/aiida/lib/python3.8/site-packages/plumpy/process_states.py", line 230, in execute
 |     result = self.run_fn(*self.args, **self.kwargs)
 |   File "/home/hollas/anaconda3/envs/aiida/lib/python3.8/site-packages/aiida/engine/processes/calcjobs/calcjob.py", line 388, in parse
 |     exit_code_retrieved = self.parse_retrieved_output(retrieved_temporary_folder)
 |   File "/home/hollas/anaconda3/envs/aiida/lib/python3.8/site-packages/aiida/engine/processes/calcjobs/calcjob.py", line 468, in parse_retrieved_output
 |     exit_code = parser.parse(**parse_kwargs)
 |   File "/home/hollas/anaconda3/envs/aiida/lib/python3.8/site-packages/aiida_orca/parsers/__init__.py", line 43, in parse
 |     parsed_obj = io.ccread(os.path.join(out_folder._repository._get_base_folder().abspath, fname_out))  #pylint: disable=protected-access
 |   File "/home/hollas/anaconda3/envs/aiida/lib/python3.8/site-packages/cclib/io/ccio.py", line 169, in ccread
 |     return log.parse()
 |   File "/home/hollas/anaconda3/envs/aiida/lib/python3.8/site-packages/cclib/parser/logfileparser.py", line 315, in parse
 |     self.extract(inputfile, line)
 |   File "/home/hollas/anaconda3/envs/aiida/lib/python3.8/site-packages/cclib/parser/orcaparser.py", line 986, in extract
 |     etenergies.append(float(broken[-2]))
 | ValueError: could not convert string to float: '='
```

I will not have time to look into it right away. I'd say if we decide to indeed use aiida and orca in my postdoc project, I could start working on this sometimes in September (or later based on priorities). For now I could perhaps at least open an issue on `cclib` repo?

**Comment by danielhollas** [comment]
I finally started to look into Aiida and ORCA again. It seems that `cclib` now supports ORCA-5.0 on master (I checked that the particular issue with TDDFT described above was fixed). Unfortunately, the fix was not included in the 1.7.1 release for some reason, looks like we need to wait for [1.7.2](https://github.com/cclib/cclib/milestone/21?closed=1). 

That being said, single point DFT calculations seem to work. When I tried optimization, I am getting the following error from the parser.
```python
 |   File "/home/hollas/Git/aiida/aiida-orca/aiida_orca/parsers/__init__.py", line 79, in parse
 |     relaxed_structure = StructureData(
 | AttributeError: module 'pymatgen' has no attribute 'Molecule'
```
I have aiida v1.6.5 and pymatgen v2022.0.10.  Looks like the Molecule class has been moved, the following import seems to fix the issue
```python
from pymatgen.core.structure import Molecule
```

Maybe to support both old and new versions we could do something like:

```python
try:
   from pymatgen.core.structure import Molecule
except:
   from pymatgen import Molecule
```

EDIT: Ignore the pymatgen stuff, it is already fixed on master, I was running old plugin version for some reason.

**Comment by danielhollas** [comment]
It looks like cclib 1.7.2 is finally close to being released

https://github.com/cclib/cclib/issues/1130

Once that is done, simply setting minimum cclib version to 1.7.2 in setup.json should do the trick.
